message-index/messages/GHC-90177/index.md

```diff
@@ -0,0 +1,13 @@
+---
+title: Orphan instance
+summary: An instance was defined separately from its type or class.
+severity: warning
+introduced: 9.6.1
+flag: -Worphans
+---
+
+When writing an instance for a typeclass, the ideal situation is that this instance is defined either in the same module as the *typeclass*, or in the same module as the *datatype*. This ensures that if a module imports either the typeclass or the datatype module, the instances will also be visible.
+
+If the instances are defined in a separate module, it becomes possible that the typeclass and the datatype are brought into scope but the *instance for that datatype* is not, leading to unexpected `No instance of Foo for type Bar`.
```

Comment:
I think it might be good to mention instance coherence here as well. I can write an example of that issue to augment your example, if you want.

Reply:
I was trying to think up an example where the solution was to wrap it in a newtype, but nothing came to mind.

Reply:
Should we just merge as is?

Reply:
Unless I'm missing something, would a test case just be:

```
data A = A

in some module
instance Eq A 
(==) _ _ = False

elsewhere
instance Eq A
(==) _ _ = True

and then a third location

case A == A of
True -> /* cool */
False -> launchNukes
```

or am I misinterpreting?

Reply:
It's annoying to rig up a system that won't complain about overlapping instances, and instead just does the wrong thing.